### ISSUE TITLE
Add option to hide specific groups

### ISSUE THREAD

**Issue by geeckon** [0xbs/premade-groups-filter#153]
Since this (https://github.com/0xbs/premade-groups-filter/issues/147) isn't possible, maybe there could be an option to quickly hide specific groups from the dungeon finder. This could be done, for example, by rightclicking the group or some combination like ctrl + rightclick. It can be quick to determine if you don't want to join a group just by looking at it, like when it isn't an upgrade for you.
There could probably also be an option to reset these hidden groups.

**Comment by 0xbs** [comment]
Not a bad idea, removal could be temporary (does not match) or permanent (WTS/spam groups). However I wonder if the issue could just be solved with a key level filter in the search box. 

Of course you might have very different ratings for individual dungeons, but isn't there always a key level where you feel comfortable that you would type into the search box? Let's say your best is Algethar +15 and Shadowmoon +18. I claim that you would not search for +16 and "click-remove" Shadowmoon +16. I think you would simply search for +18 as that's what's possible for you and provides the best loot.

A problem with this idea is that it would require quite some rewrite of the code, similar to #154. Currently groups are only removed when the list is fetched from the server. Once the list is displayed, PGF does not change it anymore. Another minor issue is that groups do not have a unique ID. So the best identification is the group leader's name, which is not immediately available.

**Comment by hollo6** [comment]
If ignoring a group is too complex to do, as an alternative, how about ignoring a leader?
Would it be more feasible to flag a player to ignore (hide) a group if they are the leader, preferably for the current session only?

**Comment by 0xbs** [comment]
I'm not sure if ignoring leaders is a good way to deal with spam, because spammers can easily create a new or use a different character that's not in your list. Most spammers groups have just one player in the group and/or no rating at all. So if your intention is to remove spam, just set a minimum rio of 1 and a minimum of two players.

The intention of this feature is to quickly remove groups that do not match your expectations for whatever reason, not really as spam removal method. When I implement this feature, I'll most likely use the new unique group ID instead of the leader name. This would mean that as soon as the leader recreates the group, it would be displayed again.

**Comment by hollo6** [comment]
What I'm trying to deal with is not spam, I have 2700 rio set as a minimum and the list is full of
- trolls who just list their groups and never invite anyone
- bait tags (e.g. they have a 10 key and write 11 12 13 14 15 in the description so their group will pop up for everyone looking in those ranges)
- scammers (promising to do the, say, 12 key after doing their 11 and they just leave after the 11).

My original idea was to ignore these groups by group ID and then when it looked like that wasn't feasible, I thought of ignoring these people by name, this way they wouldn't pop up again. I don't want to put all of them on full ignore because who knows, maybe tomorrow they will list a legit group that I'm actually interested in and there is no easy way to do so anyway.

**Comment by hollo6** [comment]
Yes, that was my thinking exactly. Is there a chance you can implement such a feature? It would be a huge QoL improvement for a lot of people. :)

**Comment by hollo6** [comment]
Awesome, thank you! Even knowing that it will be done makes me happy. :)
